va/va.h

```diff
@@ -367,10 +367,28 @@ typedef int VAStatus;   /** Return status type from functions */
 /** asynchronization,application should call additonal sync operation to access output */
 #define VA_EXEC_ASYNC             0x1
 
-/** operation mode */
+/** operation mode
+  * operation mode is defined to distinguish the code path if there are different algorithm
+  * or hw implementation candidates. application could query which implementation supported
+  * in backend driver and select one by demand.
+  * VA_EXEC_MODE_DEFAULT is the basic recommended one from backend driver, it could be same with
+  * one of VA_EXEC_MODE_XXX on different platform.
+  */
+
 #define VA_EXEC_MODE_DEFAULT      0x0
 #define VA_EXEC_MODE_POWER_SAVING 0x1
 #define VA_EXEC_MODE_PERFORMANCE  0x2
+/*
+ * VA_EXEC_MODE_3 is defined for the third solution. it is for the developer with
+ * the expertise and know enough information about the backend driver, and want to balance
+ * the hardware workload for multiple session usage. and backend driver may provide more
+ * implementation to bring in more customization/flexibility.
+ * The definition implies VA_EXEC_MODE_1 VA_EXEC_MODE_2 also is defined like
+ * #define VA_EXEC_MODE_1 VA_EXEC_MODE_POWER_SAVING
```

Comment:
I don't get the point here. You did not define neither VA_EXEC_MODE_1 nor VA_EXEC_MODE_2. Either define them or drop this comment.